javascript/javascript1/week3/smartphone.js

```diff
@@ -0,0 +1,47 @@
+////////////////////Adding activities////////////////////
+const activities = [];
+
+function addActivity(date, activity, duration) {
+  activities.push({
+    date: date,
+    activity: activity,
+    duration: duration
+  });
+}
+addActivity("23/7-18", "Youtube", 30);
+addActivity("22/7-18", "Facebook", 40);
+addActivity("22/7-18", "Whatsapp", 50);
+console.log(activities);
+
+///////////////////////Show my status///////////////////////////////
+let totalDuration = 0;
+
+function showStatus(activities) {
+  for (let i = 0; i < activities.length; i++) {
+    totalDuration = totalDuration + activities[i].duration;
+  }
+  return totalDuration;
+}
+totalDuration = showStatus(activities);
+console.log(`You have added ${activities.length} activities. They amount to ${totalDuration} min. of usage`);
+
+////////////////////////Usage limit///////////////////////////
+function maxLimit(timeLimit) {
+  if (totalDuration > timeLimit) {
```

Comment:
reusing totalDuration across multiple functions is pattern called 'shared global state', see this link for more information: https://softwareengineering.stackexchange.com/questions/148108/why-is-global-state-so-evil

This is a complex subject, but it is very important to understand if you want to build larger reliable systems. 

In short your program is working as is! So all is well. But! It is error prone. If i call the functions declared in a different order the behaviour of your program will change. 

for instance if i call 'showstatus' two times the amount will be different when i log out totalDuration, albeit this should be the same as the data behind has not changed.

Reply:
I understand. Did some changes please correct me if im wrong